transform-plugins/src/main/java/co/cask/hydrator/plugin/ValueMapper.java

```diff
@@ -0,0 +1,228 @@
+        builder = lookUpAndSetValues(input, sourceField, builder);
+        LOG.debug("in createOutputRecords after :" + builder);
+      } else {
+        // for those source field whose values are either NULL or EMPTY
+        builder.set(sourceField.getName(), "" + input.get(sourceField.getName()
+        ).toString());
+      }
+    }
+    LOG.debug("in createOutputRecords end :" + builder);
+    return builder;
+  }
+
+  /**
+   * Sets target fields and values from mapping dataset
+   */
+  private StructuredRecord.Builder lookUpAndSetValues(StructuredRecord input, Schema.Field sourceField,
+                                                      StructuredRecord.Builder builder) {
+    if (datasetMapping.containsKey(sourceField.getName())) {
+      ValueMapperLookUp valueMapperLookUp = datasetMapping.get(sourceField.getName());
+      LOG.debug("in lookUpAndSetValues:" + sourceField.getName());
+
+      if (valueMapperLookUp.lookup(input.get(sourceField.getName()).toString()) != null &&
+              valueMapperLookUp.lookup(input.get(sourceField.getName()).toString()).toString().trim().length() > 0) {
+        LOG.debug("in lookUpAndSetValues lookup value:" + valueMapperLookUp.lookup(input.get(sourceField
+                .getName
+                        ())
+                .toString()));
+        builder.set(fieldsMapping.get(sourceField.getName()),
+                valueMapperLookUp.lookup(input.get(sourceField.getName()).toString()));
+      } else {
+        builder.set(fieldsMapping.get(sourceField.getName()), "Empty");
+      }
+    } else {
+      // for those source field whose mapping is not present
+      builder.set(fieldsMapping.get(sourceField.getName()), "Empty");
+    }
+
+    return builder;
+  }
+
+  /**
+   * Creates output schema record using source and destination field mapping provided by user.
+   */
+  private Schema getOutputSchema(Schema inputSchema) throws IllegalArgumentException {
+    Schema outputSchema = schemaCache.get(inputSchema);
+    if (outputSchema != null) {
+      return outputSchema;
+    }
+    List<Schema.Field> outputFields = Lists.newArrayList();
+    for (Schema.Field inputField : inputSchema.getFields()) {
+      if (inputField.getSchema().getType() != Schema.Type.STRING) {
+        throw new IllegalArgumentException("Input field type should be String");
+      } else {
+        if (fieldsMapping.containsKey(inputField.getName())) {
+          outputFields.add(Schema.Field.of(fieldsMapping.get(inputField.getName()),
+                  Schema.of(Schema.Type.STRING)));
+        } else {
+          outputFields.add(inputField);
+        }
+      }
+    }
+    outputSchema = Schema.recordOf(inputSchema.getRecordName() + ".formatted", outputFields);
+    schemaCache.put(inputSchema, outputSchema);
+    return outputSchema;
+  }
+
+  /**
+   * Parse configuration provided by user
+   */
+  public void parseConfiguration(String mapping, TransformContext context) {
```

Comment:
there is no need to do all this manual json parsing if you are using Gson.  Just create the corresponding java objects and do `GSON.fromJson(mapping, MappingConfig.class)`